pkg/controllers/etcd.go

```diff
@@ -116,6 +116,15 @@ func (s *EtcdService) Run(ctx context.Context, ready chan<- struct{}, stopped ch
 		os.Exit(0)
 	}()
 
+	// Handle microshift-etcd termination after microshift process exits
+	defer func() {
+		cmd := exec.Command("systemctl", "stop", "microshift-etcd.scope")
+
+		if err = cmd.Start(); err != nil {
```

Comment:
`Start()` should be paired with `Wait()`. Is lack of `Wait()` a decision?
Otherwise, we can use `Run()` which will wait for completion.

How does this impact shutdown ordering?

Reply:
> `Start()` should be paired with `Wait()`. Is lack of `Wait()` a decision to make it "async and forget - don't wait for it"? Otherwise, we can use `Run()` which will wait for completion.
> 
> How does this impact shutdown ordering? Is the problem here that MicroShift starts too soon after restart? i.e. `microshift-etcd.scope` hasn't exited yet, but `microshift.service` is already up and running again?
the problem is that systemd doesnt restart `microshift-etcd.scope` although it depends on microshift.service. 
so when restart happens `microshift` is stopped but `microshift-etcd.scope` remains active, then `microshift` tries to run another instance with `systemd-run` while first is running and it fails.

Reply:
> `Start()` should be paired with `Wait()`. Is lack of `Wait()` a decision to make it "async and forget - don't wait for it"? Otherwise, we can use `Run()` which will wait for completion.

I've replaced Start() with Run() to make it synchronous.